src/service/getCreditScore.js

```diff
@@ -24,50 +60,89 @@ function getCreditScore(address) {
-                                score: accountHistoryHoldingsScore
-                            },
-                            poapsDetails: {
-                                poaps: poaps,
-                                wage: POAPS_DETAILS_WAGE,
-                                score: poapsScore
-                            },
-                            aaveAddressDetails: {
-                                details: {
-                                    borrowHistory: aaveAddressDetails.borrowHistory,
-                                    liquidationHistory: aaveAddressDetails.liquidationHistory,
-                                    repayHistory: aaveAddressDetails.repayHistory,
+                    if (isItRecurrentInvoke) {
+                        return {
+                            score: basicCreditScore,
+                            basicScore: basicCreditScore,
+                            details: {
+                                addressCreation: {
+                                    lifetimeInDays: accountLifetime.lifetimeInDays,
+                                    created: formatMomentAsDate(accountLifetime.created),
+                                    score: addressCreationScore,
+                                    wage: ADDRESS_CREATION_WAGE
+                                },
+                                tokenHoldingDetails: {
+                                    details: accountHistoryHoldings,
+                                    wage: TOKEN_HOLDING_DETAILS_WAGE,
+                                    score: accountHistoryHoldingsScore
+                                },
+                                poapsDetails: {
+                                    poaps: poaps,
+                                    wage: POAPS_DETAILS_WAGE,
+                                    score: poapsScore
                                 },
-                                wage: AAVE_ADDRESS_DETAILS_WAGE,
-                                score: aaveScore
+                                aaveAddressDetails: {
+                                    details: {
+                                        borrowHistory: aaveAddressDetails.borrowHistory,
+                                        liquidationHistory: aaveAddressDetails.liquidationHistory,
+                                        repayHistory: aaveAddressDetails.repayHistory,
+                                    },
+                                    wage: AAVE_ADDRESS_DETAILS_WAGE,
+                                    score: aaveScore
+                                }
                             }
                         }
+                    } else {
+                        return getMeanOfFriendsSocialScore(address).then(
+                            meanFriendsSocialScore => {
+                                const creditScore = (BASIC_CREDIT_SCORE * basicCreditScore) + (FRIENDS_CREDIT_SCORE * meanFriendsSocialScore);
```

Comment:
nie gadaliśmy o tym, ale jeśli dodamy followersScoreDetails z osobną wagą (o tym niżej) to suma wag będzie > 1. Mozna rozwiązać ten problem tak:
-zrobić tak żeby suma wszystkich wag włącznie z followersScoreDetails była = 1
-w przypadku kiedy liczymy credit score przyjaciół (i nie mamy followersScoreDetails) zrobić normalizację (np jeśli waga followersScoreDetails=0.2) to zrobić score * 1/0,8 (o ile dobrze to rozumiem teraz :D)

jak cos to mozemy sie zdzwonic z tym

Reply:
Nie no jest okej. Najpierw liczysz basicCreditScore który ma sumę wag == 1 i jest znormalizowany (0-1000).
Następnie liczysz średni creditScore przyjaciół który też jest znormalizowany (0-1000).

- basicCreditScore ma wagę 0.8 chyba z tego co pamiętam
- creditScore przyjaciół ma wagę 0.2

Razem sumuje się waga do jednego i nadal mamy normalizację (0-1000).
Więc imo jest git